---
title: "Infant Adiposity Differential Expression Analysis"
subtitle: "Sex Adjusted Results"
author: "Jason Laird"
date: today 
date-format: short
engine: knitr
execute: 
  warning: false
  message: false
  cache: true
format: 
  html:
    code-fold: true
    embed-resources: true
    theme: 
      - simplex
      - custom.scss
    fontcolor: '#0c051a'
---

## Setup 

### Libraries and Scripts Required

```{r,warning=FALSE, message=FALSE}
#| warning: false
#| message: false

library(tidyverse)
library(tidybulk)
library(openxlsx)
library(janitor)
library(ggpubr)
library(ggvenn)
library(scales)
library(patchwork)
library(UpSetR)
library(SummarizedExperiment)
library(clusterProfiler)
library(enrichplot)
library(EnhancedVolcano)
library(biomaRt)
library(factoextra)
library(FactoMineR)
library(sva)
library(GEOquery)
library(DESeq2)
library(edgeR)
library(DEGreport)
library(impute)
library(ggsci)

```

### Import Data

```{r}
deg_df <- readRDS("../../results/v2_results/deg_df.rds")
enrich_res_mapped <- readRDS("../../results/v2_results/enrich_res_mapped.rds")
se_filt <- readRDS("../../results/v2_results/se_filt.rds")
deg_lst <- readRDS("../../results/v2_results/deg_lst.rds")
```

## Filters

- Outliers Removed: "s34", "s459", "s29", "s72", "s307", "s349"

- nominal p-value < 0.05 & abs(log2FoldChange) > log2(1.5)

## Venn Diagram - Sex-Adjusted CNAAGs

```{r,fig.width=6,fig.height=4}
deg_lst[names(deg_lst) %in% c("Lean Sex Adj","Obese Sex Adj")] |> 
  ggvenn(
    stroke_color = "transparent",
    fill_alpha = 0.5
  )+
  coord_cartesian(clip="off")+
  scale_fill_manual(values = c("#91D1C2FF", "#8491B4FF"))
```

## Barplot of CNAAGs

```{r,fig.width=10,fig.height=4}
common_degs <- Reduce(
  intersect,
  deg_lst[names(deg_lst) %in% c(
    "Lean Sex Adj","Obese Sex Adj")] )

lean_deg <- deg_df |> 
           filter(group %in% "Lean Sex Adj") |> 
           pull(gene)

obese_deg <- deg_df |> 
  filter(group %in% "Obese Sex Adj") |> 
           pull(gene)

deg_df |> 
  filter(group %in% c("Lean Sex Adj","Obese Sex Adj")) |> 
  filter(gene %in% common_degs) |>
  mutate(group=case_when(
    group == "Lean Sex Adj" ~ "Infants with Lean Mothers",
    group == "Obese Sex Adj" ~ "Infants with Obese Mothers"
  )) |> 
  ggplot(aes(
    x = reorder(gene, -log2FoldChange),
    y = log2FoldChange,
    fill = group)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_pubr(legend = "bottom")+
  scale_fill_manual(values = c("#91D1C2FF", "#8491B4FF"))+
  rotate_x_text(angle=45)+
  labs(
    x="",
    y = expression(Log[2]~"(Fold Change)"),
    fill = "",
    title = "Common CNAAGs"
  )+
  theme(plot.title = element_text(face = "bold.italic"))
```

## Enrichment Heatmap 

```{r}
enrich_res_common <- enrichGO(
  gene =  common_degs,
        OrgDb = 'org.Hs.eg.db',
        keyType = "SYMBOL",
        ont = "ALL",
        universe = rownames(se_filt),
        pAdjustMethod = "fdr",
        pvalueCutoff = 0.1,
        qvalueCutoff = 0.1,
        maxGSSize = 800,
        minGSSize = 2
      )
```


```{r}
# identify degs that go in the same direction
same_dir_degs <- deg_df |> 
    filter(group %in% c(
        "Lean Sex Adj",
        "Obese Sex Adj")) |> 
    filter(gene %in% common_degs) |> 
  group_by(gene) |> summarise(
    direction_lean=direction[grepl("Lean",group)],
    direction_obese=direction[grepl("Obese",group)]) |>
  ungroup() |> 
  mutate(concordance=case_when(
    direction_lean==direction_obese ~ "same",
    direction_lean!=direction_obese ~ "different")) |> 
  filter(concordance=="same") |>
  pull(gene)

cnaag_dir <- deg_df |> 
    filter(group %in% c(
        "Lean Sex Adj",
        "Obese Sex Adj")) |> 
    filter(gene %in% common_degs) |> 
  group_by(gene) |> 
  summarise(
    direction_lean=direction[grepl("Lean",group)],
    direction_obese=direction[grepl("Obese",group)]) |>
  ungroup() |> 
  pivot_longer(
    cols = c(direction_lean, direction_obese),
    names_to = "group",
    values_to = "direction"
  ) |> 
  mutate(group=case_when(
    group == "direction_lean" ~ "Lean Sex Adj",
    group == "direction_obese" ~ "Obese Sex Adj"
  )) |> 
  dplyr::rename(geneID=gene)
```

```{r,fig.width=16,fig.height=6}

top_lean_terms <- enrich_res_mapped |> 
  filter(group == "Lean Sex Adj") |> 
  separate_rows(geneID, sep = "/") |> 
  filter(geneID %in% lean_deg) |> 
  dplyr::select(Description, p.adjust) |>
  distinct() |> 
  arrange(p.adjust) |>
  slice_head(n=10) |> 
  pull(Description)


top_obese_terms <- enrich_res_mapped |> 
  filter(group == "Obese Sex Adj") |> 
  separate_rows(geneID, sep = "/") |> 
  filter(geneID %in% obese_deg) |> 
  dplyr::select(Description, p.adjust) |>
  distinct() |> 
  arrange(p.adjust) |>
  slice_head(n=10) |> 
  pull(Description)

top_common_terms <- enrich_res_common |> 
  pluck("result") |> 
  arrange(p.adjust) |>
  slice_head(n=10) |> 
  pull(Description)

enrich_plot_df <- enrich_res_mapped |> 
  separate_rows(geneID, sep = "/") |> 
  filter(Description %in% c(top_lean_terms,top_obese_terms)) |> 
  filter(group %in% c(
    "Lean Sex Adj",
    "Obese Sex Adj")) |> 
  dplyr::select(Description, 
                p.adjust,
                group, 
                direction,
                geneID) |> 
  bind_rows(
    enrich_res_common |> 
  pluck("result") |> 
  separate_rows(geneID, sep = "/") |> 
  filter(Description %in% top_common_terms) |> 
  dplyr::select(Description, p.adjust,geneID) |> 
  inner_join(cnaag_dir, by="geneID") 
  )
  
enrich_plot_df |> 
  mutate(group=case_when(
    geneID %in% common_degs ~ "Common NAAGs",
    group == "Lean Sex Adj" ~ "NAAGs Unique to Lean Mothers",
    group == "Obese Sex Adj" ~ "NAAGs Unique to Obese Mothers"
  )) |> 
  ggplot(aes(
    x=geneID,
    y=fct_reorder2(Description,group,direction),
    fill=direction
  ))+
  geom_tile(alpha=.5)+
  scale_fill_manual(values = c("blue3","red4"))+
  theme_pubr(legend = "bottom")+
  rotate_x_text(angle=45)+
  labs(
    x="",
    y="",
    fill="Direction",
    title = "Functional Enrichment Results"
  )+
  facet_grid(~group,,space = "free",scales = "free")+
  theme(
    plot.title = element_text(face = "bold.italic"),
    strip.text.x = element_text(size=14, face="bold.italic"),
    axis.text.x = element_text(size=8),
    axis.text.y = element_text(size=8))
```

## Coexpression of CNAAGs


```{r,fig.width=8,fig.height=7}
# Extract log2-normalized expression for common DEGs
log2_expr_mat <- DESeqDataSet(se_filt, design = ~mfadip) |>
  estimateSizeFactors() |>
  counts(normalized = TRUE) |>
  as.data.frame() |>
  (\(df) df[rownames(df) %in% common_degs, ])() |>
  mutate(across(everything(), ~ log2(.x + 1))) |>
  as.matrix()

# Compute gene-gene correlation matrix
gene_corr_mat <- cor(t(log2_expr_mat))

#Cluster genes based on correlation distance
gene_hclust <- hclust(as.dist(1 - gene_corr_mat), method = "complete")
gene_order <- gene_hclust$labels[gene_hclust$order]

# Convert to long format for plotting
common_deg_corr_df <- gene_corr_mat |>
  as.data.frame() |>
  rownames_to_column("gene1") |>
  pivot_longer(cols = -gene1, names_to = "gene2", values_to = "corr")

# Plot clustered heatmap
common_deg_corr_df |>
  mutate(
    gene1 = factor(gene1, levels = gene_order),
    gene2 = factor(gene2, levels = gene_order)
  ) |>
  ggplot(aes(x = gene1, y = gene2, fill = corr)) +
  geom_tile() +
  theme_pubr(legend = "right") +
  scale_fill_gradient2(
    low = "#145afc",
    mid = "#ffffff",
    high = "#ee4445",
    midpoint = 0
  ) +
  rotate_x_text(angle = 90) +
  labs(
    x = "",
    y = "",
    fill = "Correlation"
  ) +
  theme(axis.ticks = element_blank())


```

## CNAAG Violin Plot By Sex

### All CNAAGs

```{r,fig.width=24,fig.height=6}
# Create a tidy data frame for plotting
log2_expr_mat_tidy <- log2_expr_mat |>
  as.data.frame() |>
  rownames_to_column("gene") |>
  pivot_longer(-gene, 
               names_to = "sample",
               values_to = "log2_expr") |>
  inner_join(colData(se_filt) |> 
               as.data.frame() |>
               rownames_to_column("sample"),
             by = "sample") |>
  filter(gene %in% common_degs)

# Plotting the violin plot
log2_expr_mat_tidy |> 
  mutate(fadip=case_when(
    fadip == "LA" ~ "Low Adiposity Neonates",
    fadip == "HA" ~ "High Adiposity Neonates"
  )) |> 
  mutate(fadip=factor(
    fadip, 
    levels=c("Low Adiposity Neonates",
             "High Adiposity Neonates"))) |> 
  ggplot(aes(
    x=fadip,
    y=log2_expr,
    fill=fadip
  ))+
  geom_violin(trim=FALSE, alpha=0.5) +
  geom_boxplot(width=0.1, outlier.shape = NA) +
  theme_pubr(legend = "right") +
  scale_fill_manual(values = c("#91D1C2FF", "#8491B4FF")) +
  facet_grid(gender~gene, 
             scales = "free_y",
             space = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.text = element_text(face = "bold.italic"))+
  labs(
    x="",
    y=expression(Log[2]~"(Expression)")
  )
```

### New CNAAGs of Interest

```{r,fig.width=8,fig.height=4}
# Plotting the violin plot
log2_expr_mat_tidy |> 
  filter(gene %in% c(c("HLA-G","IL20","MPO","DEFA1","DEFA1B"))) |> 
  mutate(fadip=case_when(
    fadip == "LA" ~ "Low Adiposity Neonates",
    fadip == "HA" ~ "High Adiposity Neonates"
  )) |>  
  mutate(fadip=factor(
    fadip, 
    levels=c("Low Adiposity Neonates",
             "High Adiposity Neonates"))) |> 
  ggplot(aes(
    x=fadip,
    y=log2_expr,
    fill=fadip
  ))+
  geom_violin(trim=FALSE, alpha=0.5) +
  geom_boxplot(width=0.1, outlier.shape = NA) +
  theme_pubr(legend = "right") +
  scale_fill_manual(values = c("#91D1C2FF", "#8491B4FF")) +
  facet_grid(gender~gene, 
             scales = "free_y",
             space = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.text = element_text(face = "bold.italic"))+
  labs(
    x="",
    y=expression(Log[2]~"(Expression)"),
    fill=""
  )
```


## CNAAG STRING

![](./cnaag_string.svg)

*PPI enrichment p-value:	9.69e-05*


## Placental Expression

```{r,fig.width=10,fig.height=8}
# bring in placental reference data
# clean data so that the cell types are in one column rather than headers
# add a column for source info
# swap short names for long cell type names
# select certain columns for plotting
# downloaded from:
# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5340963/bin/supp_gr.207597.116_Supplemental_Tables.xlsx

placenta <- openxlsx::read.xlsx(
  xlsxFile="../../data/pavlicev_2017_plancenta_reference.xlsx",
  sheet = "S1",
  startRow = 4) |>
  dplyr::select(-SE) |>
  pivot_longer(!Gene, names_to = "Cell type", values_to = "nTPM") |>
  rename("Gene name"="Gene") |>
  mutate(log2_exp = log2(nTPM+1)) |>
  mutate(tissue="Placenta") |> 
  filter(`Gene name` %in% common_degs) |>
  dplyr::select(
    `Gene name`,`Cell type`,log2_exp,tissue
  ) |> 
  mutate(
    `Cell type`=stringi::stri_replace_all_regex(
      `Cell type`,
      pattern=c("CYT1" ,"CYT2" ,"CYT3" ,"DC"   ,"EVT"  ,"SYN" ,"ESF"  ,"DEC"),
      replacement = c("Intravillous cytotrophoblast 1",
                      "Intravillous cytotrophoblast 2",
                      "Intravillous cytotrophoblast 3",
                      "Dendritic Cell",
                      "Extravillous Trophoblast",
                      "Syncytiotrophoblast",
                      "Endometrial Stromal Fibroblast",
                      "Decidual Cell"),
      vectorize_all = FALSE
    )
  )

# convert counts to log2 exp so we can compare the two references on the same scale
# add a column for source info
# select certain columns for plotting
hpa_immune = read_tsv(
  "../../data/rna_immune_cell.tsv"
) |>
  mutate(log2_exp = log2(nTPM+1)) |>
  mutate(tissue="Human Protein Atlas - Immune") |>
  rename("Cell type"="Immune cell") |>
  filter(`Gene name` %in% common_degs) |>
  dplyr::select(
    `Gene name`,`Cell type`,log2_exp,tissue
  ) 

# rank gene expression by placental reference
placenta_rank = placenta |>
  group_by(`Gene name`) |>
  summarise(placenta_rank=mean(log2_exp))

# merge two sources into one df for plotting
placenta_hpa_immune <- rbind(placenta,hpa_immune) |>
  filter(`Gene name` %in% common_degs) |>
  inner_join(placenta |>
               group_by(`Gene name`) |>
               summarise(placenta_rank=mean(log2_exp)),
             by=c("Gene name"="Gene name")) 

# plot the expression of common adiposity drivers in both
# placental reference data and the hpa immune cell reference
placenta_hpa_immune |>
  ggplot(aes(
    x=reorder(`Cell type`,log2_exp),
    y=reorder(`Gene name`,placenta_rank),
    size=log2_exp,
    color=tissue
  ))+
  geom_point()+
  theme_pubr(legend = "right")+
  scale_color_manual(values=c(
    "Human Protein Atlas - Immune"=get_palette("npg",k=10)[1],
    "Placenta"=get_palette("npg",k=10)[6]
  ))+
  guides(color="none")+
  scale_size(range=c(.1,10))+
  rotate_x_text(angle=45)+
  facet_grid(.~tissue,
             scales = "free",
             space="free")+
  theme(strip.text.x.top = element_text(face = "bold.italic",
                                        size = 12))+
  labs(
    x="",
    y="",
    size=expression(Log[2]~"(TPM + 1)"),
  )
```

